### PR TITLE
Merge esql branch into master

### PR DESCRIPTION
**Pull request by pquentin**
Now that https://github.com/elastic/rally-tracks/pull/385 was merged, the tests pass and we can merge the esql branch into master.

### CODE REVIEW

**Comment by pquentin** [comment]
The CLA check was failing because @ywelsch left the company and had pushed some commits to this branch with his @elastic.co email address. This is a known issue with the check, it cannot tell when some commits were made (commit dates can be forged after all) but we know those were made while Yannick was still an employee, so we're good.

To make CLA happy, I rebased and made myself author of those commits with Yannick co-author, and then force pushed. (Force pushing the esql branch was probably not a great move, sorry.)

**Comment by pquentin** [comment]
@csoulios Would you like to add a description stating that this is a work-in-progress in this pull request or in a follow-up?

File: nyc_taxis/challenges/default.json [hunk 1]
```diff
@@ -667,4 +667,232 @@
           "iterations": 50
         }
       ]
+    },
+    {
+      "name": "esql",
+      "description": "Performance benchmarks for the ESQL compute engine",
```
Comment (csoulios) [inline nyc_taxis/challenges/default.json:673]:
```suggestion
      "description": "Performance benchmarks for internal R&D on query languages. This is work in progress",
```

**Review by csoulios — APPROVED** [review]
LGTM! 

Thanks for merging this

**Pull request merged**